File: algomap/javascript/easy_2239_findClosestNumberToZero.ts
function findClosestNumber(nums: number[]): number {
  let closest = nums[0];

  for (let num of nums) {
    const currDist = Math.abs(0 - num);

    if (
      currDist < Math.abs(closest) ||
      (currDist === Math.abs(closest) && num > closest)
    ) {
      closest = num;
    }
  }

  return closest;
}

const cases = [-4, -2, 1, 4, 8];
console.log(findClosestNumber(cases));

File: algomap/python/easy_2239_find_closest_number_to_zero.py
from typing import List

class Solution:
    def findClosestNumber(self, nums: List[int]) -> int:
        return min(nums, key=lambda x: (abs(x), -x))
    

solution = Solution()

test = [-4,-2,1,4,8]

print(solution.findClosestNumber(test))